lib/impure/db_sqlite.nim

```diff
@@ -716,8 +719,60 @@ proc setEncoding*(connection: DbConn, encoding: string): bool {.
 proc finalize*(sqlPrepared:SqlPrepared){.discardable.} = 
   discard finalize(sqlPrepared.PStmt)
 
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: int32):int =
+  ## Binds a int32  to the specified paramIndex.
+  result = bind_int(ps.PStmt, paramIdx.int32, val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int,val: int64):int =
+  ## Binds a int64  to the specified paramIndex.
+  result = bind_int64(ps.PStmt, paramIdx.int32, val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: float64):int =
+  ## Binds a 64bit float to the specified paramIndex.
+  result = bind_double(ps.PStmt, paramIdx.int32, val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int):int =
+  ## Sets the bindparam at the specified paramIndex to null 
+  ## (default behaviour by sqlite).
+  result = bind_null(ps.PStmt, paramIdx.int32) 
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int,val: string): int =
+  ## Binds a string to the specified paramIndex.
+  result = bind_text(ps.PStmt, paramIdx.int32,val.cstring,-1.int32 , SQLITE_STATIC)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int,val: cstring): int =
+  ## binds a blob to the specified paramIndex.
+  let len = val.len
+  result = bind_blob(ps.PStmt, paramIdx.int32, val[0].unSafeAddr, len.int32 , SQLITE_STATIC)
+
+proc bindParams*[T](ps: SqlPrepared, params:var T) = 
+  var idx:int = 1
+  for v in params.fields:
+    when v is int:
+      if SQLITE_OK != ps.bindParam(idx,v.int32):
+        discard
+    elif v is int32:
+        if SQLITE_OK != ps.bindParam(idx,v):
+          discard
+    elif v is int64:
+        if SQLITE_OK != ps.bindParam(idx,v):
+          discard
+    elif v is float64:
+        if SQLITE_OK != ps.bindParam(idx,v):
+          discard
+    elif v is string:
+        if SQLITE_OK != ps.bindParam(idx,v):
+          discard
+    elif v is cString:
+        if SQLITE_OK != ps.bindParam(idx,v):
+          discard
+    else:
+      discard
+    inc idx
```

Comment:
If done this way, then only one Nim parameter can be bound. Instead, turn this into a macro that takes a `vararg[untyped]`, then just generate code that calls `bindParam()` on each entry.

The syntax I have in mind is like this:

`statement.bindParams(10, "foo", 42.0)`

Which should generates:

```nim
  statement.bindParam(10, 0)
  statement.bindParam("foo", 1)
  statement.bindParam(42.0, 2)
```

IMO this would feel more intuitive.

Reply:
I turn it to template and handle iterable. make logic simple, otherwise I should considering `bindParams(10,[2,3],3)`